flocker/provision/_install.py

```diff
@@ -1532,6 +1581,90 @@ def configure_cluster(
     ])
 
 
+def reinstall_flocker_at_version(
+    reactor, nodes, control_node, package_source, distribution
+):
```

Comment:
So I copied the approach from there, and they use common code for uninstalling and re-installing flocker.

The one difference is the ManagedRunner generates new certificates every time, whereas I just want to restart the flocker agents.

It might make sense for the managed runner to start with a fresh cluster_id so that there aren't old volumes polluting the test, whereas this specifically wants to use the same cluster_id. I think we're hitting a reasonable amount of codeshare, although there definitely would be room for improvement.